RecoEgamma/PhotonIdentification/plugins/PhotonMVAEstimatorRunIIFall17.cc

```diff
@@ -0,0 +1,312 @@
+  // This should work for both reco and pat.
+  const edm::Ptr<reco::Photon> phoRecoPtr(particle);
+  if( phoRecoPtr.isNull() )
+    throw cms::Exception("MVA failure: ")
+      << " given particle is expected to be reco::Photon or pat::Photon," << std::endl
+      << " but appears to be neither" << std::endl;
+  
+  // Both pat and reco particles have exactly the same accessors.
+  auto superCluster = phoRecoPtr->superCluster();
+  // Full 5x5 cluster shapes. We could take some of this directly from
+  // the photon object, but some of these are not available.
+  float e2x2 = std::numeric_limits<float>::max();
+  float e5x5 = std::numeric_limits<float>::max();
+  float full5x5_sigmaIetaIeta = std::numeric_limits<float>::max();
+  float full5x5_sigmaIetaIphi = std::numeric_limits<float>::max();
+  float effSigmaRR = std::numeric_limits<float>::max();
+  
+  AllVariables allMVAVars;
+  
+  const auto& full5x5_pss = phoRecoPtr->full5x5_showerShapeVariables();
+  e2x2 = full5x5_pss.e2x2;
+  e5x5 = full5x5_pss.e5x5;
+  full5x5_sigmaIetaIeta = full5x5_pss.sigmaIetaIeta;
+  full5x5_sigmaIetaIphi = full5x5_pss.sigmaIetaIphi;
+  effSigmaRR = full5x5_pss.effSigmaRR;
+  
+  allMVAVars.varRawE         = superCluster->rawEnergy(); 
+  allMVAVars.varR9           = phoRecoPtr->r9() ;
+  allMVAVars.varSieie        = full5x5_sigmaIetaIeta; 
+  allMVAVars.varSCEtaWidth   = superCluster->etaWidth(); 
+  allMVAVars.varSCPhiWidth   = superCluster->phiWidth(); 
+  allMVAVars.varSieip        = full5x5_sigmaIetaIphi;
+  allMVAVars.varE2x2overE5x5 = e2x2/e5x5;
+  allMVAVars.varPhoIsoRaw    = (*phoPhotonIsolationMap)[phoRecoPtr];  
+  allMVAVars.varChIsoRaw     = (*phoChargedIsolationMap)[phoRecoPtr];
+  allMVAVars.varWorstChRaw   = (*phoWorstChargedIsolationMap)[phoRecoPtr];
+  allMVAVars.varSCEta        = superCluster->eta(); 
+  allMVAVars.varRho          = *rho; 
+  allMVAVars.varESEffSigmaRR = effSigmaRR;
+  allMVAVars.varESEnOverRawE = superCluster->preshowerEnergy() / superCluster->rawEnergy();
+  
+  constrainMVAVariables(allMVAVars);
+  //
+  // Important: the order of variables in the "vars" vector has to be EXACTLY
+  // the same as in the .xml file defining the MVA.
+  //
+  std::vector<float> vars;
+  if( isEndcapCategory( findCategory( particle ) ) ) {
+    vars = packMVAVariables(
+                                       allMVAVars.varRawE,
+                                       allMVAVars.varR9,
+                                       allMVAVars.varSieie,
+                                       allMVAVars.varSCEtaWidth,
+                                       allMVAVars.varSCPhiWidth,
+                                       allMVAVars.varSieip,
+                                       allMVAVars.varE2x2overE5x5,
+                                       allMVAVars.varPhoIsoRaw,
+				       allMVAVars.varChIsoRaw,
+				       allMVAVars.varWorstChRaw,
+                                       allMVAVars.varSCEta,
+				       allMVAVars.varRho,
```

Comment:
the indentation is inconsistent and is probably too far to the right compared to the open brace in L255.

Reply:
Changes implemented here:
https://github.com/lsoffi/cmssw/blob/CMSSW_9_4_0_pre3_TnP/RecoEgamma/PhotonIdentification/plugins/PhotonMVAEstimatorRunIIFall17.cc